### PR TITLE
To Do list: list structure

### PR DESCRIPTION
**Pull request by vicmaburrito**
- [x]  Set up a new project with webpack that is based on the webpack exercise you have already completed.
- [x] Create an index.html file and write your HTML markup here. 
- [x] Create an empty To-Do List placeholder 
     The index.html file must be set as a template using the HTML Webpack Plugin.
     Create an index.js file and set an array of some simple to do tasks (array of objects). Each task object should contain three 
     keys:
     description [string].
     completed [bool].
     index: [number].
- [x] Write a function to iterate over the tasks array and populate an HTML list item element for each task.
- [x] On page load render the dynamically created list of tasks in the dedicated placeholder. The list appear in order of 
      the index values for each task.
- [x] Create a style.css and set rules for the To Do List. CSSloaded by Webpack Style/CSS Loader.

### CODE REVIEW

File: dist/index.html [hunk 1]
```diff
@@ -1,12 +1,33 @@
...
-</head>
-<body>
-    <script src="bundle.js"></script>
-</body>
+    <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+    <link href="https://fonts.googleapis.com/css2?family=Open+Sans+Condensed:ital,wght@1,300&display=swap" rel="stylesheet">
+    <script defer src="main.js"></script>
+  <script defer src="main.js"></script></head>
```
Comment (D4nielJ) [inline dist/index.html:12]: - [x] Let's remove the script from your index.html template in `src`. As you can see webpack already adds this script automatically to the compiled `html`. This is causing your script run twice and the tasks are being added twice to the list.

image

File: src/index.html [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+<html lang="en">
+  <head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>TO-DO List</title>
+    <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+    <link href="https://fonts.googleapis.com/css2?family=Open+Sans+Condensed:ital,wght@1,300&display=swap" rel="stylesheet">
+    <script defer src="main.js"></script>
```
Comment (D4nielJ) [inline src/index.html]: [OPTIONAL] I think the best option for you right now to add Bootstrap to the project, is by simply adding the `cdn` link to the head of your file. Although this isn't the best alternative for production-ready projects, it is a really good alternative for light-weight projects.

You will find out as you advance in the program, that other tools come with webpack-ready builds where you can import libraries more easily.

If you still want to add bootstrap manually you could run `npm install bootstrap`, and then import `import 'bootstrap/dist/css/bootstrap.css'` to the top of your index.js file. (This is very similar to just importing the CDN script)

If you want more control over which parts of bootstrap you want in your project, you need to set up a pipeline to import SASS. [You can read more about it here.](https://getbootstrap.com/docs/5.1/getting-started/download/) [And here.](https://github.com/twbs/bootstrap-npm-starter)

**Review by D4nielJ — CHANGES_REQUESTED** [review]
## Hello @vicmaburrito, 

You're doing great! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights :fire: :fire: 
- Your project is well structured!
- I liked the way you defined your todos tasks! :fireworks:

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

## Happy coding and stay awesome! 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by helman101** [comment]
# STATUS: APPROVED 🟢 🥳 

Hi @vicmaburrito 🥇 , 

### Congratulations!!! 🥳 🎉

You have successfully completed all the expectations for this project, there is nothing more to do here, now you can go on your way and keep improving your skills.

Cheers and Happy coding!👏👏👏

gif

If anything is not 100% clear please leave your questions or comments in the PR thread, we will be glad to help you 😄.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**